spec/abci/abci.md

```diff
@@ -393,10 +402,10 @@ via light client.
 * **Usage**:
 
     * Technically optional - not involved in processing blocks.
-    * Guardian of the mempool: every node runs CheckTx before letting a
+    * Guardian of the mempool: every node runs `CheckTx` before letting a
     transaction into its local mempool.
     * The transaction may come from an external user or another node
-    * CheckTx need not execute the transaction in full, but rather a light-weight
+    * `CheckTx` need not execute the transaction in full, but rather a light-weight
     yet stateful validation, like checking signatures and account balances, but
     not running code in a virtual machine.
```

Comment:
It need not, but I think we perhaps want a stronger statement: We seem to _rely_ on the fact that checking a transaction doesn't actually apply any state changes. If that's correct, then "stateful" here may be doing a bit more work than it should. What I think this is trying to say is something like

> `CheckTx` validates the transaction against the current state of the application, for example checking signatures and account balances, but does not apply any state changes described in the transaction.

Does that seem right? Presumably the application could actually run the transaction against a non-persistent clone of the current state to see if it's valid (an implementation detail) but it seems like "non-persistent" is important.